### ISSUE TITLE
[universal-redux] setToken example

### ISSUE THREAD

**Issue by karthikiyengar** [diegohaz/arc#189]
Hi,
Thank you for the wonderful boilerplate. It's been a great learning experience, just looking at the code.

For authenticated server side requests, here I have the external REST endpoint set a httpOnly cookie upon login. The universal-server can now use that cookie to prefetch other authenticated data before render and set the logged in state in my reducer.

So my fetching logic in pages looks like this (ideally an action will be invoked, capturing data in the store)
```
import api from 'services/api'
...
  static get({ store, req }) {
    return new Promise((resolve, reject) => {
      api.post('/endpoint', { data: 'Boom' }, { headers: { Authorization: req.cookies.token } })
          .then(resolve)
          .catch(reject)
    })
  }
```
This does seem to work fine and authenticated requests are made, but I am wondering how I could use setToken to eliminate having to pass the token manually each time.

An example would be appreciated. Additionally, is this the intended way to use the fetching logic? Thanks.
 
P.S: I have thought about api.create on server.js, using setToken there, and passing the instance as a parameter to the static methods. But this approach would not work for calls made on the client side (since I'd be directly importing from services/api)

**Comment by diegohaz** [comment]
Hey, @karthikiyengar 

It's better to call the API in a saga and just dispatch an action in the `get` method.

You can pass down the API object through the sagas and just set the token on the entry point. Take a look at this: https://github.com/diegohaz/arc/blob/feature/redux-saga-test-plan/src/index.js#L14

To share state (token) between server and client, see #166

**Comment by diegohaz** [comment]
Closed in #217
